src/nni_manager/core/nniDataStore.ts

```diff
@@ -358,35 +339,94 @@ class NNIDataStore implements DataStore {
-                            hParamIds.add(newHParam.parameter_index);
-                            hParamIdMap.set(jobInfo.id, hParamIds);
+                const hyperParameter: any = this.parseHyperParameter(record.data);
+                if (hyperParameter !== undefined) {
+                    // update trials here.
+                    const trialId = this.getTrialId(jobInfo.id, hyperParameter);
+                    let trialInfo: TrialJobInfo | undefined;
+                    if (trialMap.has(trialId)) {
+                        trialInfo = trialMap.get(trialId);
+                        if (trialInfo && trialInfo.hyperParameters) {
+                            trialInfo.hyperParameters.push(JSON.stringify(hyperParameter));
                         }
                     } else {
-                        assert(false, 'jobInfo.hyperParameters is undefined');
+                        trialInfo = {
+                            id: trialId,
+                            status: this.getJobStatusByLatestEvent('RUNNING', record.event),
+                            hyperParameters: [JSON.stringify(hyperParameter)],
+                            startTime: record.timestamp,
+                            sequenceId: sequenceId++
+                        };
+                        trialMap.set(trialId, trialInfo);
                     }
+                    if (!trialInfo) {
+                        throw new Error('Empty trialInfo');
+                    }
+
+                    let trialIds = jobTrialMap.get(jobInfo.id);
+                    if (trialIds === undefined) {
+                        trialIds = [];
+                        jobTrialMap.set(jobInfo.id, trialIds);
+                    }
+                    trialIds.push(trialId);
                 }
             }
-            if (record.sequenceId !== undefined && jobInfo.sequenceId === undefined) {
-                jobInfo.sequenceId = record.sequenceId;
+        }
+
+        for (const jobTrial of jobTrialMap) {
+            const jobInfo = jobMap.get(jobTrial[0]);
+            const trialIds = jobTrial[1];
+            let trialInfo: TrialJobInfo | undefined;
+            let lastTrialInfo: TrialJobInfo | undefined;
+
+            if (!jobInfo) {
+                throw new Error('Empty jobInfo');
+            }
+
+            for (const trialId of trialIds) {
+                trialInfo = trialMap.get(trialId);
+                if (!trialInfo) {
+                    throw new Error('Empty trialInfo');
+                }
+                if (lastTrialInfo) {
+                    lastTrialInfo.endTime = trialInfo.startTime;
+                }
+                trialInfo.logPath = jobInfo.logPath;
+                // There is no status reported for each trial. So assume it's success, if it's not last one.
+                // The last trial status is the same as job.
+                trialInfo.status = 'SUCCEEDED';
```

Comment:
what if a parameter does not have a matched final result, due to catched error by user?

Reply:
As it needs metric data record to parse this kind of problem, and it's high performance impact in this function.